overlord/snapstate/snapstate.go

```diff
@@ -307,17 +307,21 @@ func Install(st *state.State, name, channel string, revision snap.Revision, user
 		return nil, &snap.AlreadyInstalledError{Snap: name}
 	}
 
-	snapInfo, err := snapInfo(st, name, channel, revision, userID, flags)
+	info, err := snapInfo(st, name, channel, revision, userID)
 	if err != nil {
 		return nil, err
 	}
 
+	if !validInfoForFlags(info, &snapst, flags) {
+		return nil, fmt.Errorf("cannot find snap %q", name)
```

Comment:
Just to be sure, is that the message you're going to tweak later?  We need to be more clear about the fact it exists rather than claiming it's not found.